### PR TITLE
fixed floating point error

### PR DESCRIPTION
**Pull request by michaelpkuhn**
Tests failed due to floating point. We address this by rounding floating points when calculated. However, some issues seem to persist at present.

### CODE REVIEW

File: tests/resources/census_poverty_dump_expected_output.json [hunk 1]
```diff
@@ -897,17 +897,17 @@
...
+                        1474.6,
                         2136.4,
                         4294.199999999999,
                         11028.60000000001,
                         778020.0
                     ],
                     "poverty_population_poverty_child": [
                         124.0,
-                        414.20000000000005,
+                        414.2000000000001,
```
Comment (michaelpkuhn) [inline tests/resources/census_poverty_dump_expected_output.json:908]: This is still an issue

File: tests/resources/census_poverty_dump_expected_output.json [hunk 2]
```diff
@@ -897,17 +897,17 @@
                     },
                     "poverty_population_poverty": [
                         482.0,
-                        1474.6000000000001,
+                        1474.6,
```
Comment (michaelpkuhn) [inline tests/resources/census_poverty_dump_expected_output.json:900]: Fixed it here though...

File: src/data.py [hunk 3]
```diff
@@ -145,9 +145,11 @@ def calculate_quantiles_bins(df: pd.DataFrame,
     columns_to_drop = [column for column in df.columns
                        if column not in columns_to_bin]
 
-    quantile_df = df.drop(columns_to_drop, axis=1)
+    quantile_df = df.drop(columns_to_drop, axis=1).astype('float64')
 
     quantile_df = quantile_df.apply(numpy.quantile, q=make_quantiles_q(bin_count))  # noqa: E501
     # round for space and avoid floating point imprecision
-    quantile_df = numpy.round(quantile_df, 6)
-    return quantile_df.to_dict(orient='list')
```
Comment (michaelpkuhn) [inline src/data.py:153]: Broken up to assist debugging

File: src/data.py [hunk 4]
```diff
@@ -145,9 +145,11 @@ def calculate_quantiles_bins(df: pd.DataFrame,
     columns_to_drop = [column for column in df.columns
                        if column not in columns_to_bin]
 
-    quantile_df = df.drop(columns_to_drop, axis=1)
+    quantile_df = df.drop(columns_to_drop, axis=1).astype('float64')
```
Comment (michaelpkuhn) [inline src/data.py:148]: This fixed the issue

File: src/data.py (hunk 3 above)
Comment (michaelpkuhn) [reply src/data.py:153]: quantile_df was made of `objects` not floats, so `.to_dict()` introduced impercision

**Comment by michaelpkuhn** [comment]
When we calculated quantile bins, we don't check the type. So when a df of `object` is passed, when you convert it `to_dict` it introduces imprecision by casting those objects to `float`.

This solution casts the passed df to `float64` before processing.

**Review by brickman1444 — APPROVED** [review]
Nice job! Everything looks good.

**Pull request merged**